Source/BugsnagEvent.m

```diff
@@ -315,13 +320,14 @@ - (instancetype)initWithKSReport:(NSDictionary *)report {
             _releaseStage = [report valueForKeyPath:@"user.state.oom.app.releaseStage"];
             _handledState = [BugsnagHandledState handledStateWithSeverityReason:LikelyOutOfMemory];
             _deviceAppHash = [report valueForKeyPath:@"user.state.oom.device.id"];
-            _metadata = [NSMutableDictionary new];
+            self.metadata = [BugsnagMetadata new];
+            
             NSDictionary *sessionData = [report valueForKeyPath:@"user.state.oom.session"];
             if (sessionData) {
                 _session = [[BugsnagSession alloc] initWithDictionary:sessionData];
                 _session.unhandledCount += 1; // include own event
                 if (_session.user) {
-                    _metadata = @{@"user": [_session.user toJson]};
+                    self.metadata = [[BugsnagMetadata alloc] initWithDictionary:[@{@"user": [_session.user toJson]} mutableCopy]];
```

Comment:
nit: unnecessary mutableCopy

Reply:
Actually, I think it's required since the signature is:

`- (id)initWithDictionary:(NSMutableDictionary *)dict`

Reply:
Could the constructor be altered to use `NSDictionary`? This would allow both types to be passed in and avoid a need for a copy.

Reply:
You're right, if we change the sig. and I think we can without loss of functionality.  We make no promises as long as _some_ form of NS*Dictionary is passed in.